samza-kafka/src/main/scala/org/apache/samza/config/KafkaConfig.scala

```diff
@@ -338,6 +340,15 @@ class KafkaConfig(config: Config) extends ScalaMapConfig(config) {
 
     kafkaChangeLogProperties.setProperty("segment.bytes", KafkaConfig.CHANGELOG_DEFAULT_SEGMENT_SIZE)
     kafkaChangeLogProperties.setProperty("delete.retention.ms", String.valueOf(new StorageConfig(config).getChangeLogDeleteRetentionInMs(name)))
+
+    // To enable transactional state, we will need to avoid the head of the changelog
+    // (the messages after last checkpoint) being log-compacted so we can trim the rest of the updates.
+    // We use min.compaction.log.ms to control the compaction time.
+    if (new TaskConfig(this).getTransactionalStateRestoreEnabled) {
+      kafkaChangeLogProperties.setProperty("min.compaction.lag.ms",
```

Comment:
Nitpick: extract constant for config key.

Reply:
Fixed.